pkg/util/pki/pki.go

```diff
@@ -0,0 +1,105 @@
+package pki
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+)
+
+var caMap map[string]x509.CertPool
```

Comment:
I would add create a struct that has these. In `initCaMap()`, you can ensure the struct gets created only once.

Reply:
"these" being caMap and mu

Reply:
don't you just need
```go
var caMap = make(map[string]x509.CertPool)
```